jtests/DateTimeTest.java

```diff
@@ -0,0 +1,243 @@
+        assertTrue(dt1.__le__(dt2).toBoolean());
+        assertTrue(dt1.__lt__(dt2).toBoolean());
+        assertFalse(dt1.__ge__(dt2).toBoolean());
+        assertFalse(dt1.__gt__(dt2).toBoolean());
+
+        org.python.Object[] args3 = {Int.getInt(4), Int.getInt(2), Int.getInt(5)};
+        DateTime dt3 = new DateTime(args3, kwargs);
+
+        assertFalse(dt3.__eq__(dt1).toBoolean());
+        assertTrue(dt3.__ne__(dt1).toBoolean());
+        assertFalse(dt3.__le__(dt1).toBoolean());
+        assertFalse(dt3.__lt__(dt1).toBoolean());
+        assertTrue(dt3.__ge__(dt1).toBoolean());
+        assertTrue(dt3.__gt__(dt1).toBoolean());
+    }
+
+    @Test
+    public void testTimestamp() {
+        Map<String, org.python.Object> kwargs = new HashMap<String, org.python.Object>();
+        org.python.Object[] args = {
+            Int.getInt(1970),
+            Int.getInt(1),
+            Int.getInt(1),
+            Int.getInt(1),
+            Int.getInt(1),
+            Int.getInt(1),
+            Int.getInt(0),
+        };
+        DateTime dt1 = new DateTime(args, kwargs);
+        double ts = ((org.python.types.Float) dt1.timestamp()).value;
+
+        //TODO: This test will be incorrect if run in another timezone
+        assertEquals((long)ts, 61);
+    }
+
+    @Test
+    public void testFromIsoString() {
+        DateTime dt = (DateTime) DateTime.fromisoformat("1970-01-02T03:04:05.000010");
+
+        assertEquals(dt.year, Int.getInt(1970));
+        assertEquals(dt.month, Int.getInt(1));
+        assertEquals(dt.day, Int.getInt(2));
+        assertEquals(dt.hour, Int.getInt(3));
+        assertEquals(dt.minute, Int.getInt(4));
+        assertEquals(dt.second, Int.getInt(5));
+        assertEquals(dt.microsecond, Int.getInt(10));
+    }
+
+    @Test
+    public void testWeekday() {
+        Map<String, org.python.Object> kwargs = new HashMap<String, org.python.Object>();
+        org.python.Object[] args = {
+            Int.getInt(2020),
+            Int.getInt(9),
+            Int.getInt(28),
+        };
+        DateTime dt = new DateTime(args, kwargs);
+        assertEquals(dt.weekday(),Int.getInt(0));
+    }
+    
+    
+    @Test
+    public void testReturns() {
+        Map<String, org.python.Object> kwargs = new HashMap<String, org.python.Object>();
+        org.python.Object[] args = {
+            Int.getInt(2020),
+            Int.getInt(9),
+            Int.getInt(28),
+            Int.getInt(1),
+            Int.getInt(2),
+            Int.getInt(3),
+            Int.getInt(4),
+        };
+        DateTime dt = new DateTime(args, kwargs);
+
+        assertEquals(dt.year, dt.__year__());
+        assertEquals(dt.month, dt.__month__());
+        assertEquals(dt.day, dt.__day__());
+        assertEquals(dt.hour, dt.__hour__());
+        assertEquals(dt.minute, dt.__minute__());
+        assertEquals(dt.second, dt.__second__());
+        assertEquals(dt.microsecond, dt.__microsecond__());
+    }
```

Comment:
Add tests for date edge-cases like end of month etc.

Reply:
I do not quite follow you. The tests here are not used for edgecases/test of values. They are used to check whether the __minute__,__second__ .. methods are working as intended. :)

Reply:
"__minute__","__second__" (apparently double underscore makes it bold)

Reply:
\_\_minute\_\_ , \_\_second\_\_ **

Reply:
Sorry for being unclear, I meant that tests for those edge cases is needed but not necessarly in this particular place :D

Reply:
Ah, alright! :D